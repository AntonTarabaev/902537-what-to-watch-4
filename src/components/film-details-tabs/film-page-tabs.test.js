import FilmPageTabs from "@root/components/film-details-tabs/film-page-tabs";

const film = {
  id: `5593482`,
  releaseYear: 2001,
  genre: `Comedy`,
  rating: 10,
  ratingVotes: 19,
  director: `Green One`,
  description: [
    `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
    `Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`,
    `Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`,
  ],
  actors: [
    `Robert Rodrigues`,
    `Takeshi Kitano`,
    `Michael Caine`,
    `Robert De Niro`,
    `Tom Hanks`,
    `Takeshi Kitano`,
    `Michael Caine`,
  ],
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  duration: 123,
  comments: [
    {
      author: `Quentin Tarantino`,
      comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Sed sed nisi sed augue convallis suscipit in sed felis.. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`,
      date: new Date(4827394),
      rating: 1.7,
    },
    {
      author: `Tom Hanks`,
      comment: `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra. Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.`,
      date: new Date(89321),
      rating: 2.3,
    },
    {
      author: `Michael Caine`,
      comment: `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
      date: new Date(983928),
      rating: 3.8,
    }
  ]
};

describe(`FilmDetailsTabs component render correctly`, () => {
  it(`on Overview tab`, () => {
    const tree = renderer.create(
        <FilmPageTabs
          film={film}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`on Details tab`, () => {
    const component = renderer.create(
        <FilmPageTabs
          film={film}
        />
    );

    component.root.findAllByProps({className: `movie-nav__link`})[1].props.onClick({preventDefault: () => {}});

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`on Reviews tab`, () => {
    const component = renderer.create(
        <FilmPageTabs
          film={film}
        />
    );

    component.root.findAllByProps({className: `movie-nav__link`})[2].props.onClick({preventDefault: () => {}});

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
