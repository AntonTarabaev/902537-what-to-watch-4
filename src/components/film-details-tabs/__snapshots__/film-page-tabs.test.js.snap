// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetailsTabs component render correctly on Details tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href={null}
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Green One
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Robert Rodrigues
          ,
          <br />
          Takeshi Kitano
          ,
          <br />
          Michael Caine
          ,
          <br />
          Robert De Niro
          ,
          <br />
          Tom Hanks
          ,
          <br />
          Takeshi Kitano
          ,
          <br />
          Michael Caine
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          2h 3m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Comedy
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2001
        </span>
      </p>
    </div>
  </div>,
]
`;

exports[`FilmDetailsTabs component render correctly on Overview tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href={null}
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      10
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Awesome
      </span>
      <span
        className="movie-rating__count"
      >
        19
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
    </p>
    <p>
      Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.
    </p>
    <p>
      Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Green One
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Robert Rodrigues, Takeshi Kitano, Michael Caine, Robert De Niro and others
      </strong>
    </p>
  </div>,
]
`;

exports[`FilmDetailsTabs component render correctly on Reviews tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href={null}
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Sed sed nisi sed augue convallis suscipit in sed felis.. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Quentin Tarantino
            </cite>
            <time
              className="review__date"
              dateTime="1970-01-01"
            >
              January 1, 1970
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          1.7
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra. Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Tom Hanks
            </cite>
            <time
              className="review__date"
              dateTime="1970-01-01"
            >
              January 1, 1970
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          2.3
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Michael Caine
            </cite>
            <time
              className="review__date"
              dateTime="1970-01-01"
            >
              January 1, 1970
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          3.8
        </div>
      </div>
    </div>
  </div>,
]
`;
